Lib/traceback.py

```diff
@@ -621,3 +624,83 @@ def format(self, *, chain=True):
             yield 'Traceback (most recent call last):\n'
         yield from self.stack.format()
         yield from self.format_exception_only()
+
+
+class TracebackExceptionGroup:
+    """An exception or exception group ready for rendering.
+
+    The traceback module captures enough attributes from the original exception
+    group to this intermediary form to ensure that no references are held, while
+    still being able to fully print or format it.
+
+    Use `from_exception` to create TracebackExceptionGroup instances from exception
+    objects, or the constructor to create TracebackExceptionGroup instances from
+    individual components.
+
+    - :attr:`summary` A tuple of (indentation level, TracebackException) pairs
+    """
+
+    def __init__(self, exc_type, exc_value, exc_traceback, **kwargs):
+        self.summary = tuple(
+            self._gen_summary(
+                exc_type, exc_value, exc_traceback, **kwargs))
+        self._str = _some_str(exc_value)
+
+    def _gen_summary(self, exc_type, exc_value, exc_traceback,
+                     indent=0, **kwargs):
+        # recursively extract the sequence of (indent, TracebackException)
+        # pairs. If exc_value is a single exception, there is only one pair
+        # (with indent 0). If it is an exception group, we get one pair
+        # for each exception in it, showing the whole tree
+        te = TracebackException(
+            exc_type, exc_value, exc_traceback, **kwargs)
+        yield tuple((indent, te))
+        if isinstance(exc_value, exception_group.ExceptionGroup):
+            for e in exc_value.excs:
+                yield from self._gen_summary(
+                    type(e), e, e.__traceback__, indent=indent+4, **kwargs)
+
+    @classmethod
+    def from_exception(cls, exc, *args, **kwargs):
+        """Create a TracebackException from an exception."""
+        return cls(type(exc), exc, exc.__traceback__, *args, **kwargs)
+
+    def format(self, *, chain=True):
```

Comment:
Shouldn't this be implemented in terms of format_exception_only()? See TracebackException.format().

Reply:
This just calls TracebackException.format for each exception and lets it do what it needs to do.

Reply:
Oh I see. Right.